optimade/server/config.py

```diff
@@ -22,46 +23,72 @@ class ServerConfig(BaseSettings):
 
     """
 
-    debug: bool = False
-    use_real_mongo: bool = False
-    mongo_database: str = "optimade"
-    mongo_uri: str = "localhost:27017"
-    links_collection: str = "links"
-    references_collection: str = "references"
-    structures_collection: str = "structures"
-    page_limit: int = 20
-    page_limit_max: int = 500
-    default_db: str = "test_server"
-    base_url: Optional[str] = None
+    config_file: str = Field(
+        "~/.optimade.json", description="File to load alternative defaults from"
+    )
+    debug: bool = Field(
+        False, description="Turns on Debug Mode for the Optimade Server implementation"
+    )
+    use_real_mongo: bool = Field(
+        False, description="Use a real Mongo server rather than MongoMock"
+    )
+    mongo_database: str = Field(
+        "optimade", description="Mongo database for collection data"
+    )
+    mongo_uri: str = Field("localhost:27017", description="URI for the Mongo server")
+    links_collection: str = Field("links", description="Collection name for Links")
+    references_collection: str = Field(
+        "references", description="Collection name for References"
+    )
+    structures_collection: str = Field(
+        "structures", description="Collection name for Structures"
+    )
+    page_limit: int = Field(20, description="Default items per page")
+    page_limit_max: int = Field(500, description="Max items per page")
+    default_db: str = Field(
+        "test_server",
+        description="ID of the provider's chosen default OPTIMADE API database.",
+    )
+    base_url: Optional[str] = Field(
+        None, description="URL for the homepage for this implementation"
+    )
     implementation: Implementation = Implementation(
         name="Example implementation",
         version=__version__,
         source_url="https://github.com/Materials-Consortia/optimade-python-tools",
         maintainer=None,
     )
     provider: Provider = Provider(
-        prefix="_exmpl_",
+        prefix="exmpl",
         name="Example provider",
         description="Provider used for examples, not to be assigned to a real database",
         homepage="https://example.com",
         index_base_url="http://localhost:5001",
     )
-    provider_fields: Dict[Literal["links", "references", "structures"], List[str]] = {
-        "structures": ["band_gap", "_mp_chemsys"]
-    }
-    aliases: Dict[Literal["links", "references", "structures"], Dict[str, str]] = {
-        "structures": {
-            "id": "task_id",
-            "chemical_formula_descriptive": "pretty_formula",
-            "chemical_formula_reduced": "pretty_formula",
-            "chemical_formula_anonymous": "formula_anonymous",
-        }
-    }
+    provider_fields: Dict[Literal["links", "references", "structures"], List[str]] = {}
```

Comment:
```suggestion
    provider_fields: Dict[Literal["links", "references", "structures"], List[str]] = Field({}, description="A list of additional fields to be served with the provider's prefix attached, broken down by endpoint.")
```

Might need to double check that the default value of `{}` works here